### PR TITLE
feat : leaderboard

### PR DESCRIPTION
**Pull request by rohit8686**

### CODE REVIEW

**Review by RahulSingh9131 — APPROVED** [review]
* clean and readable code
* css seems fine too
 Go ahead and merge.

**Review by Divya70 — APPROVED** [review]
clean code 
everything looks good to me , you can merge

File: src/frontend/pages/LeaderBoard/LeaderBoard.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  const {
+    quizState: { quizCategory },
+  } = useQuiz();
+  const [usersQuizData, setUsersQuizData] = useState([]);
+  let leaderBoardData = [...usersQuizData].sort(
+    (a, b) => b.data[quizCategory]?.score - a.data[quizCategory]?.score
+  );
+
+  console.log("leaderboard");
+  console.log(leaderBoardData);
```
Comment (Logan1x) [inline src/frontend/pages/LeaderBoard/LeaderBoard.jsx]: You may want to remove these comments

File: src/frontend/Routing/Routing.jsx [hunk 2]
```diff
@@ -22,6 +23,7 @@ export const Routing = () => {
         <Route path="/quiz/rules/:quizName" element={<Rules />} />
         <Route path="/results" element={<Results />} />
         <Route path="/profile" element={<UserProfile />} />
+        <Route path="/leaderboard" element={<LeaderBoard />} />
       </Route>
```
Comment (Logan1x) [inline src/frontend/Routing/Routing.jsx:29]: Can add 404 error page

**Review by Logan1x — APPROVED** [review]
- Good use of Firebase
- code quality and naming conventions are good.
- If there are o breaking changes. It's good to merge.

File: src/frontend/Routing/Routing.jsx (hunk 2 above)
Comment (rohit8686) [reply src/frontend/Routing/Routing.jsx:29]: yeah will add it

**Pull request merged**